trigger/netdevices/__init__.py

```diff
@@ -492,6 +502,57 @@ def os(self):
                     Otherwise template does not exist.""")
             return None
 
+    def _get_endpoint(self, *args):
+        endpoint = generate_endpoint(self).wait()
+        # factory = Factory()
+        factory = TriggerEndpointClientFactory()
+        factory.protocol = IoslikeSendExpect
+        # prompt = re.compile(settings.DEFAULT_PROMPT_PAT)
+        prompt = re.compile('R1>')
+        self._connected = True
+        return endpoint.connect(factory, prompt_pattern=prompt)
+        
+    def open(self):
+        def inject_net_device_into_protocol(proto):
+            proto.net_device = self
+            return proto
+
+        self._endpoint = self._get_endpoint()
+        self.d = self._endpoint.addCallback(
+                inject_net_device_into_protocol
+                )
+        return True
+
+    def close(self):
+        def disconnect(proto):
+            proto.loseConnection()
```

Comment:
This should be `proto.transport.loseConnection()`. The `NameError` is blocking results from coming back.